src/plugins/auto_batch/CMakeLists.txt

```
@@ -0,0 +1,20 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "ov_auto_batch_plugin")
+
+file(GLOB SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+
+file(GLOB HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/*.hpp)
+
+ie_add_plugin(NAME ${TARGET_NAME}
+              DEVICE_NAME "BATCH"
+              SOURCES ${SOURCES} ${HEADERS}
+              VERSION_DEFINES_FOR auto_batch.cpp)
+
+target_link_libraries(${TARGET_NAME} PRIVATE ngraph inference_engine_transformations Threads::Threads)
```

Why do we need threads here?
```suggestion
target_link_libraries(${TARGET_NAME} PRIVATE Threads::Threads)
```

---

Maybe will be better to use:
```
set_ie_threading_interface_for(${TARGET_NAME})
```

---

no, this doesn't work, as the code needs the pthreads (for std::therad), just like the ov_runime :)